fluent/src/types.rs

```diff
@@ -40,10 +45,32 @@ impl FluentValue {
                 (a - b).abs() < f32::EPSILON
             }
             (&FluentValue::String(ref a), &FluentValue::Number(ref b)) => {
-                //XXX: This is a dirty hack and should be replaced with a
-                //lazy resolved cache on the context.
-                let pr = PluralRules::new(ctx.locales);
-                pr.select(*b) == a
+                // //XXX: This is a dirty hack and should be replaced with a
+                // //lazy resolved cache on the context.
+
+                // ^ This has been replaced with the code below.
+
+                let cat = match a.as_str() {
+                    "zero" => PluralCategory::ZERO,
+                    "one" => PluralCategory::ONE,
+                    "two" => PluralCategory::TWO,
+                    "few" => PluralCategory::FEW,
+                    "many" => PluralCategory::MANY,
+                    "other" => PluralCategory::OTHER,
+                    _ => return false,
+                };
+
+                let supported = negotiate_languages(
+                    ctx.locales,
+                    IntlPluralRules::get_locales(PluralRuleType::CARDINAL),
+                    Some("en"),
+                    &NegotiationStrategy::Lookup,
+                );
+
+                let locale = supported[0].to_owned();
+
+                let pr = IntlPluralRules::create(&locale, PluralRuleType::CARDINAL).unwrap();
```

Comment:
Turning a string slice into an owner String, just to pass a reference (slice) to it in the next line doesn't make much sense ;)
Next time:

```
let pr = IntlPluralRules::create(supported[0], ...);
```